Subject: ntdll: preserve native service context across server suspension

--- a/dlls/ntdll/unix/thread.c
+++ b/dlls/ntdll/unix/thread.c
@@ -1078,7 +1078,7 @@
         context_to_server( &server_contexts[count++], native_machine, native_context, native_machine );
         if (wow_context) context_to_server( &server_contexts[count++], main_image_info.Machine,
                                             wow_context, main_image_info.Machine );
-        else if (native_machine != main_image_info.Machine)
+        else if (!get_thread_data()->native_domain && native_machine != main_image_info.Machine)
             context_to_server( &server_contexts[count++], main_image_info.Machine,
                                native_context, native_machine );
     }
@@ -1103,7 +1103,7 @@
         context_from_server( native_context, &server_contexts[0], native_machine );
         if (wow_context)
             context_from_server( wow_context, &server_contexts[1], main_image_info.Machine );
-        else
+        else if (!get_thread_data()->native_domain)
             context_from_server( native_context, &server_contexts[1], native_machine );
     }
     else context_from_server( wow_context, &server_contexts[0], main_image_info.Machine );
